### ISSUE TITLE
Change Request: Warn if no rules are configured

### ISSUE THREAD

**Issue by Wilfred** [eslint/eslint#7431]
**The version of ESLint you are using.**

3.8.0

**The problem you want to solve.**

eslint does not warn the user if no rules are configured.

This is helpful when a user accidentally specifies rules at the top level of their config:

``` json
{
    "indent": ["warn", 4],
    "semi": "warn"
}
```

**Your take on the correct solution to problem.**

I think that `--debug` should report the number of rules processed, e.g.

```
  eslint:config Loading /home/wilfred/projects/metawiki/.eslintrc.json +3ms
  eslint:config Loaded 0 rules from confing  +0ms
```

I think eslint should also log a warning in this case, even when `--debug` isn't specified.

**Comment by mysticatea** [comment]
Thank you for the issue.

Sounds good to me.
ESLint fails if there is no configuration. I think nice if ESLint behaves the same if the found configuration has no valid property.

**Comment by kaicataldo** [comment]
@eslint/eslint-team This seems like a "nice to have". Thoughts on putting a "help wanted" label on this?

**Comment by not-an-aardvark** [comment]
How should the warning be displayed? Would it be available to a formatter, or would it just be logged to stderr?

**Comment by nzakas** [comment]
I'm not sure if this is a good idea or not. People may choose to use ESLint just to parse code without enabling rules, and I'm not sure we'd want to display a warning by default because of that. 

I would support some type of config validation, though, that would throw an error if an invalid top-level key is found. (What @mysticatea said.)

**Comment by kaicataldo** [comment]
Closing this as discussion has stalled and it doesn't appear there is enough interest, either from the team or from the community, to continue the discussion at this time. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. Please feel free to open a new issue with a proposal to continue this discussion if this is something you would like to continue advocating for.
